packages/desktop-client/src/components/reports/reports/CashFlowCard.tsx

```diff
@@ -104,19 +261,74 @@ export function CashFlowCard({
   const [start, end] = calculateTimeRange(meta?.timeFrame, defaultTimeFrame);
   const [nameMenuOpen, setNameMenuOpen] = useState(false);
 
-  const params = useMemo(
-    () => simpleCashFlow(start, end, meta?.conditions, meta?.conditionsOp),
-    [start, end, meta?.conditions, meta?.conditionsOp],
-  );
-  const data = useReport('cash_flow_simple', params);
+  const isConcise = (() => {
+    const numDays = d.differenceInCalendarDays(
+      d.parseISO(end),
+      d.parseISO(start),
+    );
+    return numDays > 31 * 3;
+  })();
 
   const [isCardHovered, setIsCardHovered] = useState(false);
   const onCardHover = useCallback(() => setIsCardHovered(true), []);
   const onCardHoverEnd = useCallback(() => setIsCardHovered(false), []);
 
-  const { graphData } = data || {};
-  const expenses = -(graphData?.expense || 0);
-  const income = graphData?.income || 0;
+  const paramsCondensed = useMemo(
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏it feels like a lot of logic in this component is now directly duplicated from `CashFlow.tsx`. Could you instead create a new component that both `CashFlow` and this component could use directly? To reduce the duplication.

Reply:
I see, but I'm busy now, so I pushed the other changes please check if I was in the right way.